src/annotator/highlighter.ts

```diff
@@ -234,10 +234,16 @@ export function highlightRange(
   // inserting highlight elements in places that can only contain a restricted
   // subset of nodes such as table rows and lists.
   const whitespace = /^\s*$/;
-  textNodeSpans = textNodeSpans.filter(span =>
-    // Check for at least one text node with non-space content.
-    span.some(node => !whitespace.test(node.data)),
-  );
+  textNodeSpans = textNodeSpans.filter(span => {
+    const parentElement = span[0].parentElement;
+    return (
+      // Whitespace <span>s should be highlighted since they affect layout
+      (parentElement?.childNodes.length === 1 &&
```

Comment:
Two reasons: First I wanted to make this as tight as possible to avoid accidentally catching unexpected cases. Second, allowing multi-child would allow this case, inter-element whitespace:

```html
<span>
    <span>Foo</span>
    [WHITESPACE]
    <span>Bar</span>
</span>
```

(Note newlines/indent in actual html would themselves constitute whitespace, but I'm just adding them for clarity, pretend it's all one line for this and the rest of the examples)

Note this is whitespace _between_ two elements in a span parent. This whitespace would be captured if I didn't include the single-child check. I imagine this type of whitespace is likely sliiiightly less intentional than the case I am handling:

```html
<span>
    <span>Foo</span>
    <span>[WHITESPACE]</span>
    <span>Bar</span>
</span>
```

Note browsers also collapse these whitespaces when rendering, so if you have:

```html
<span>
    <span>Foo</span>
    [WHITESPACE]
    <span>[WHITESPACE]</span>
    [WHITESPACE]
    <span>Bar</span>
</span>
```

It'll be displayed as `Foo Bar` with only one space in the middle. I wasn't entirely sure what impact that sort of logic might have, so decided to avoid the question entirely and use the single-child check :P Happy to revisit if y'all have a better suggestion! Note that some code syntax highlighters (Codemirror, GitHub) use inter-element whitespace (which currently does not get wrapped in the hypothesis `<mark>` equivalent), not whitespace-only spans (what BookReader, code snippets on hypothesis docs use). This solution only addresses the latter cases.